src/CourseDashboard.cpp

```diff
@@ -2,23 +2,28 @@
 #include <stdexcept>
 #include <iostream>
 #include <algorithm>
+#include <fstream>
+#include <string>
 
-CourseDashboard::CourseDashboard() {}
 
-void CourseDashboard::showAll()
+CourseDashboard::CourseDashboard() = default;
+
+std::string CourseDashboard::showAll()
 {
+    std::string result;
     for (auto & user : users_)
     {
-        std::cout << user.getAllInfo();
+        result.append(user.getAllInfo());
```

Comment:
ostringstream could be a bit more efficient instead of append

Reply:
Python way :/ :)